### DISCUSSION TITLE
Hide invalid Clearnet Lightning URI on Cloudflare Tunneled Instance.

### DISCUSSION THREAD

**Discussion by gobrrrme** [btcpayserver/btcpayserver#5893] [Ideas / Feature Requests]
Hi everyone,

as discussed in a previous issue which I can't find anymore, lightning nodes can't be reached through their clearnet URI if the BTCPay instance is connected through a Cloudflare Tunnel. I tested many tunnel configurations, and it's just not possible to reach port 9735. Cloudflare confirmed this, not extensively, but they confirmed. The discussion back then made it clear that there is no way around this, but maybe to limit it's impact.

When opening the "Node Info" screen or looking at URIs in RTL or lncligetinfo, both URIs are being displayed. I think it would be a great quality of life improvement if the clearnet URI would not be shown at all if the Cloudflare tunnel is used. This could prevent many conversations like "I can't connect to your node, are you sure it's online" and would lead to people picking the correct URI by default.

Just a quality of life improvement for Noderunners.

Cheers.

**Comment by dennisreimann** [comment]
@NicolasDorier Is there a way we can passthrough the info that the node is running via Cloudflare?

Basically this is the combination Internal node + [Cloudflare Docker fragment](https://github.com/btcpayserver/btcpayserver-docker/blob/master/docker-compose-generator/docker-fragments/opt-add-cloudflared.yml). If we have that info available in the BTCPay Server container, then we can simply remove the clearnet connection information from the public node page.

**Comment by NicolasDorier** [comment]
@dennisreimann I don't think we should make a configuration settings for this. We will end up with people scratching their head as to why the link is onion.

What can be done though is a setting at the Lightning payment method configuration level which force the use of onion for exposing lightning.

**Reply by Jean681729** [reply]
Ok👍
